src/c/lib.nim

```diff
@@ -0,0 +1,8 @@
+import ../lib
+import sys
+
+proc hashMessage*(message: cstring): cstring =
+  let hash = lib.hashMessage($message)
+  result = cast[cstring](c_malloc(csize_t hash.len + 1))
+  copyMem(result, hash.cstring, hash.len)
+  result[hash.len] = '\0'
```

Comment:
Why is it necessary to do a malloc + copyMem instead of returning the result of hashMessage directly?

Reply:
Because `hash` (or `hash.cstring`) will eventually be garbage collected by Nim, whereas memory allocated with `c_malloc` won't be, thus leaving it up to the caller to free the memory occupied by the return value. That's the same principle as with `status-go` — even though Go is a gc'd language, the return values of the status-go methods aren't garbage collected (as I understood from chatting with @cammellos).

We don't need to do such a thing in `src/lib.nim` because it's assumed that the consumer will be a program implemented in Nim. But in the case of `src/c/lib.nim` we assume that the consumer will be e.g. a program written in C, linking against the nim sources compiled into a library.

Reply:
I should note:  what I explained above is my understanding of the situation, but I could be mistaken and it's unnecessary.